<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Seungjun Ahn" />


<title>A User Manual for PRANA</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">A User Manual for PRANA</h1>
<h4 class="author">Seungjun Ahn</h4>
<h4 class="date">July 22nd, 2024 (Version 1.0.5)</h4>



<p>We introduce a Pseudo-value Regression Approach for Network Analysis (PRANA) [1]. To our knowledge, this is the first attempt of utilizing a regression modeling for the differential network (DN) analysis by collective gene expression levels under two experimental conditions (<span class="math inline">\(\textit{e.g.}\)</span> ‘current’ vs. ‘non-current smokers’ or ‘high-risk’ vs. ‘low-risk’). We start from the mutual information (MI) criteria, followed by pseudo-value calculations, which are then entered into a robust regrssion model.</p>
<div id="requirements" class="section level2">
<h2>Requirements</h2>
<p>Please install these R packages prior to use PRANA. Note that minet package is available in Bioconductor. Please see below for further instruction. Of important note (as of July 2024), dnapath R package is archived in CRAN as of May 2024, which no longer becomes a requirement for the installation of PRANA. Instead, run_aracne() function is available in our package as of version 1.0.5 to obtain mutual information (MI) estimate via ARACNE for network estimation step.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># library(dplyr) </span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># library(parallel) # To use mclapply() when reestimating the association matrix.</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># library(robustbase) # To fit a robust regression</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># library(minet) # To estimate network via ARACNE</span></span>
<span id="cb1-5"><a href="#cb1-5"></a></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co"># Please run the following lines to install minet package </span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># from Bioconductor in your R console:</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co">#if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co">#    install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co">#BiocManager::install(&quot;minet&quot;)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(PRANA)</span></code></pre></div>
<div id="example-load-the-copdgene-study-data-from-prana-r-package" class="section level3">
<h3>Example: Load the COPDGene study data from PRANA R package:</h3>
<p>In the original article [1], a real-data analysis was performed to showcase the utility of PRANA. This contains clinical and expression data for 406 samples and 28 COPD-related genes that were highlighted in a recent genome-wide association study [2]. The full data is available from the Gene Expression Database with accession number GSE158699 [3].</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">data</span>(combinedCOPDdat_RGO) <span class="co"># A complete data containing expression and clinical data.</span></span></code></pre></div>
</div>
<div id="data-processing-for-the-example-data-from-copdgene-study" class="section level3">
<h3>Data processing for the example data from COPDGene study:</h3>
<p>The main variable of our interest in this analysis is the current smoking status. We obtain the indices of subjects who are ‘current’ vs. ‘non-current smokers.’ These indices are used to dichotomize expression dataset into ‘current (Group B)’ and ‘non-current smokers (Group A).’ This is important as the we estimate the group-specific <span class="math inline">\(p \times p\)</span> association matrices.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># A gene expression data part of the downloaded data.</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>rnaseqdat &lt;-<span class="st"> </span>combinedCOPDdat_RGO[ , <span class="dv">8</span><span class="op">:</span><span class="kw">ncol</span>(combinedCOPDdat_RGO)]</span>
<span id="cb4-3"><a href="#cb4-3"></a>rnaseqdat &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">apply</span>(rnaseqdat, <span class="dv">2</span>, as.numeric))</span>
<span id="cb4-4"><a href="#cb4-4"></a></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co"># A clinical data with additional covariates sorted by current smoking groups:</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co"># The first column is ID, so do not include.</span></span>
<span id="cb4-7"><a href="#cb4-7"></a>phenodat &lt;-<span class="st"> </span>combinedCOPDdat_RGO[<span class="kw">order</span>(combinedCOPDdat_RGO<span class="op">$</span>currentsmoking), <span class="dv">2</span><span class="op">:</span><span class="dv">7</span>]</span>
<span id="cb4-8"><a href="#cb4-8"></a></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co"># Indices of non-current smoker (namely Group A)</span></span>
<span id="cb4-10"><a href="#cb4-10"></a>index_grpA &lt;-<span class="st"> </span><span class="kw">which</span>(combinedCOPDdat_RGO<span class="op">$</span>currentsmoking <span class="op">==</span><span class="st"> </span><span class="dv">0</span>)</span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co"># Indices of current smoker (namely Group B)</span></span>
<span id="cb4-12"><a href="#cb4-12"></a>index_grpB &lt;-<span class="st"> </span><span class="kw">which</span>(combinedCOPDdat_RGO<span class="op">$</span>currentsmoking <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="apply-the-prana" class="section level3">
<h3>Apply the PRANA:</h3>
<p>Once the data processing is done, we are off to apply the PRANA. In <code>PRANA</code> function, please make sure you specify the expression and clinical data separately. Additionally, you will need to provide the indices for each group of a binary indicator variable that is deemed as a main predictor variable.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>PRANAres &lt;-<span class="st"> </span><span class="kw">PRANA</span>(<span class="dt">RNASeqdat =</span> rnaseqdat, <span class="dt">clindat =</span> phenodat, <span class="dt">groupA =</span> index_grpA, <span class="dt">groupB =</span> index_grpB)</span></code></pre></div>
</div>
<div id="some-supporting-features-after-the-use-of-prana" class="section level3">
<h3>Some supporting features after the use of PRANA:</h3>
<p>In this package, there are some additional R functions that can help you locating the name of genes that are significantly differentially connected (DC) and names and adjusted p-values via the empirical Bayes adjustment [4] for all variables and for a specific variable.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="co"># This is useful when we want to create a table with adjusted p-values only.</span></span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">adjpval</span>(PRANAres)</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co">#&gt;        currentsmoking packyrs   age gender  race FEV1perc</span></span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="co">#&gt; 10370           0.004   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co">#&gt; 10420           0.000   1.000 0.998      1 1.000        1</span></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">#&gt; 1306            0.686   0.996 1.000      1 1.000        1</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">#&gt; 155185          0.000   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">#&gt; 158158          0.846   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">#&gt; 1653            0.002   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">#&gt; 1762            0.000   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">#&gt; 23389           0.020   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-12"><a href="#cb6-12"></a><span class="co">#&gt; 253461          0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-13"><a href="#cb6-13"></a><span class="co">#&gt; 26112           0.800   0.870 1.000      1 1.000        1</span></span>
<span id="cb6-14"><a href="#cb6-14"></a><span class="co">#&gt; 27436           0.000   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-15"><a href="#cb6-15"></a><span class="co">#&gt; 3308            0.000   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-16"><a href="#cb6-16"></a><span class="co">#&gt; 3696            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-17"><a href="#cb6-17"></a><span class="co">#&gt; 374739          0.000   1.000 0.996      1 0.626        1</span></span>
<span id="cb6-18"><a href="#cb6-18"></a><span class="co">#&gt; 3842            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-19"><a href="#cb6-19"></a><span class="co">#&gt; 406             0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-20"><a href="#cb6-20"></a><span class="co">#&gt; 56986           0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-21"><a href="#cb6-21"></a><span class="co">#&gt; 57188           0.000   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-22"><a href="#cb6-22"></a><span class="co">#&gt; 6239            0.462   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-23"><a href="#cb6-23"></a><span class="co">#&gt; 7067            0.000   0.998 0.996      1 0.330        1</span></span>
<span id="cb6-24"><a href="#cb6-24"></a><span class="co">#&gt; 7871            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-25"><a href="#cb6-25"></a><span class="co">#&gt; 79961           0.004   1.000 1.000      1 1.000        1</span></span>
<span id="cb6-26"><a href="#cb6-26"></a><span class="co">#&gt; 79991           0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-27"><a href="#cb6-27"></a><span class="co">#&gt; 8224            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-28"><a href="#cb6-28"></a><span class="co">#&gt; 8853            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-29"><a href="#cb6-29"></a><span class="co">#&gt; 8870            0.000   1.000 0.996      1 1.000        1</span></span>
<span id="cb6-30"><a href="#cb6-30"></a><span class="co">#&gt; 9258            0.000   1.000 0.996      1 0.994        1</span></span>
<span id="cb6-31"><a href="#cb6-31"></a><span class="co">#&gt; 9686            0.846   0.972 0.996      1 1.000        1</span></span></code></pre></div>
<p>Back to the description of the original article by Ahn <span class="math inline">\(\textit{et al.}\)</span>, the main interest is in the current smoking status to declare whether a gene is DC between current and non-current smokers at the user-specified significance level. Thus, we subset the vector of p-values for current smoking status from the <code>adjpvaltab</code> object, aka an object with adjusted p-values and gene names using <code>adjpval</code> function above.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># Create an object to keep the table with adjusted p-values using adjpval() function.</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>adjptab &lt;-<span class="st"> </span><span class="kw">adjpval</span>(PRANAres)</span></code></pre></div>
<p><code>adjpval_specific_var</code> is a handy function to retrieve adjusted p-values with gene names for a single variable of interest.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: Please do NOT forget to provide a name of variable with the quotation marks!</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">adjpval_specific_var</span>(<span class="dt">adjptab =</span> adjptab, <span class="dt">varname =</span> <span class="st">&quot;currentsmoking&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="co">#&gt;        currentsmoking</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#&gt; 10370           0.004</span></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co">#&gt; 10420           0.000</span></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="co">#&gt; 1306            0.686</span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co">#&gt; 155185          0.000</span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#&gt; 158158          0.846</span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="co">#&gt; 1653            0.002</span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#&gt; 1762            0.000</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="co">#&gt; 23389           0.020</span></span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="co">#&gt; 253461          0.000</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#&gt; 26112           0.800</span></span>
<span id="cb8-14"><a href="#cb8-14"></a><span class="co">#&gt; 27436           0.000</span></span>
<span id="cb8-15"><a href="#cb8-15"></a><span class="co">#&gt; 3308            0.000</span></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="co">#&gt; 3696            0.000</span></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="co">#&gt; 374739          0.000</span></span>
<span id="cb8-18"><a href="#cb8-18"></a><span class="co">#&gt; 3842            0.000</span></span>
<span id="cb8-19"><a href="#cb8-19"></a><span class="co">#&gt; 406             0.000</span></span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="co">#&gt; 56986           0.000</span></span>
<span id="cb8-21"><a href="#cb8-21"></a><span class="co">#&gt; 57188           0.000</span></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="co">#&gt; 6239            0.462</span></span>
<span id="cb8-23"><a href="#cb8-23"></a><span class="co">#&gt; 7067            0.000</span></span>
<span id="cb8-24"><a href="#cb8-24"></a><span class="co">#&gt; 7871            0.000</span></span>
<span id="cb8-25"><a href="#cb8-25"></a><span class="co">#&gt; 79961           0.004</span></span>
<span id="cb8-26"><a href="#cb8-26"></a><span class="co">#&gt; 79991           0.000</span></span>
<span id="cb8-27"><a href="#cb8-27"></a><span class="co">#&gt; 8224            0.000</span></span>
<span id="cb8-28"><a href="#cb8-28"></a><span class="co">#&gt; 8853            0.000</span></span>
<span id="cb8-29"><a href="#cb8-29"></a><span class="co">#&gt; 8870            0.000</span></span>
<span id="cb8-30"><a href="#cb8-30"></a><span class="co">#&gt; 9258            0.000</span></span>
<span id="cb8-31"><a href="#cb8-31"></a><span class="co">#&gt; 9686            0.846</span></span></code></pre></div>
<p>If you would like to quickly know which genes are significantly DC for your main binary grouping variable, please use <code>sigDCGtab</code> to proceed. This will return a table with adjusted p-values and gene names.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: Please do NOT forget to provide a name of variable with the quotation marks!</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>sigDCGtab &lt;-<span class="st"> </span><span class="kw">sigDCGtab</span>(<span class="dt">adjptab =</span> adjptab, <span class="dt">groupvar =</span> <span class="st">&quot;currentsmoking&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>sigDCGtab</span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co">#&gt;        currentsmoking</span></span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="co">#&gt; 10370           0.004</span></span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="co">#&gt; 10420           0.000</span></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="co">#&gt; 155185          0.000</span></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co">#&gt; 1653            0.002</span></span>
<span id="cb9-9"><a href="#cb9-9"></a><span class="co">#&gt; 1762            0.000</span></span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co">#&gt; 23389           0.020</span></span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="co">#&gt; 253461          0.000</span></span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co">#&gt; 27436           0.000</span></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="co">#&gt; 3308            0.000</span></span>
<span id="cb9-14"><a href="#cb9-14"></a><span class="co">#&gt; 3696            0.000</span></span>
<span id="cb9-15"><a href="#cb9-15"></a><span class="co">#&gt; 374739          0.000</span></span>
<span id="cb9-16"><a href="#cb9-16"></a><span class="co">#&gt; 3842            0.000</span></span>
<span id="cb9-17"><a href="#cb9-17"></a><span class="co">#&gt; 406             0.000</span></span>
<span id="cb9-18"><a href="#cb9-18"></a><span class="co">#&gt; 56986           0.000</span></span>
<span id="cb9-19"><a href="#cb9-19"></a><span class="co">#&gt; 57188           0.000</span></span>
<span id="cb9-20"><a href="#cb9-20"></a><span class="co">#&gt; 7067            0.000</span></span>
<span id="cb9-21"><a href="#cb9-21"></a><span class="co">#&gt; 7871            0.000</span></span>
<span id="cb9-22"><a href="#cb9-22"></a><span class="co">#&gt; 79961           0.004</span></span>
<span id="cb9-23"><a href="#cb9-23"></a><span class="co">#&gt; 79991           0.000</span></span>
<span id="cb9-24"><a href="#cb9-24"></a><span class="co">#&gt; 8224            0.000</span></span>
<span id="cb9-25"><a href="#cb9-25"></a><span class="co">#&gt; 8853            0.000</span></span>
<span id="cb9-26"><a href="#cb9-26"></a><span class="co">#&gt; 8870            0.000</span></span>
<span id="cb9-27"><a href="#cb9-27"></a><span class="co">#&gt; 9258            0.000</span></span></code></pre></div>
<p>Lastly, a function below, called <code>sigDCGnames</code>, will be useful when you just want to return the names of the significantly DC genes from PRANA at the 0.05 significance level.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: Please do NOT forget to provide a name of variable with the quotation marks!</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>sigDCGnames &lt;-<span class="st"> </span><span class="kw">sigDCGnames</span>(<span class="dt">adjptab =</span> adjptab, <span class="dt">groupvar =</span> <span class="st">&quot;currentsmoking&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb10-3"><a href="#cb10-3"></a>sigDCGnames</span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="co">#&gt;  [1] &quot;10370&quot;  &quot;10420&quot;  &quot;155185&quot; &quot;1653&quot;   &quot;1762&quot;   &quot;23389&quot;  &quot;253461&quot; &quot;27436&quot; </span></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="co">#&gt;  [9] &quot;3308&quot;   &quot;3696&quot;   &quot;374739&quot; &quot;3842&quot;   &quot;406&quot;    &quot;56986&quot;  &quot;57188&quot;  &quot;7067&quot;  </span></span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="co">#&gt; [17] &quot;7871&quot;   &quot;79961&quot;  &quot;79991&quot;  &quot;8224&quot;   &quot;8853&quot;   &quot;8870&quot;   &quot;9258&quot;</span></span></code></pre></div>
<p>As an additional step, we can use <code>rename_genes</code> function to rename Entrez gene IDs into gene symbols. However, it is currently not available since <code>dnapath</code> package is currently archived. A user may manually install the archived version of <code>dnapath</code> from the CRAN, and follow the R code below.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">#rename_genes(sigDCGnames, to = &quot;symbol&quot;, species = &quot;human&quot;)</span></span></code></pre></div>
<div style="page-break-after: always;"></div>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>[1] Ahn S, Grimes T, Datta S. (2023). A pseudo-value regression approach for differential network analysis of co-expression data. <span class="math inline">\(\textit{BMC Bioinformatics}\)</span>. <strong>24</strong>(1), 8.</p>
<p>[2] Sakornsakolpat P, Prokopenko D, Lamontagne M, and et al. (2019). Genetic landscape of chronic obstructive pulmonary disease identifies heterogeneous cell-type and phenotype associations. <span class="math inline">\(\textit{Nature Genetics}\)</span>, <strong>51</strong>(3), 494–505.</p>
<p>[3] Wang Z, Masoomi A, Xu Z, and et al. (2021). Improved prediction of smoking status via isoform-aware RNA-seq deep learning models. <span class="math inline">\(\textit{PLoS Computational Biology}\)</span>, <strong>17</strong>(10), e1009433.</p>
<p>[4] Datta S, Datta S. (2005). Empirical Bayes screening of many p-values with applications to microarray studies. <span class="math inline">\(\textit{Bioinformatics}\)</span>, <strong>21</strong>(9), 1987–1994.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
